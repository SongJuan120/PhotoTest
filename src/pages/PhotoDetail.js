import React from 'react'
import { useLocation } from "react-router-dom";
import '../assets/styles/style.css'


const PhotoDetail = () => {
    const location = useLocation();
    const item = location.state.item

    return (       
        <div className='detail-main'>            
            <div className='user-container'>
                <h2>Name: {item.user}</h2>                
            </div>
            <img src={item.largeImageURL} width='80%' />       
            <h2>downloads: {item.downloads}</h2>                                             
            <h2>views: {item.views}</h2>                     
            <h2>size: {item.imageSize}</h2>           
            <h2>Tags: {item.tags}</h2>
        </div>            
    )
}
export default PhotoDetail
