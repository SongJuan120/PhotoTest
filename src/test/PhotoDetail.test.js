import React from 'react';
import renderer from 'react-test-renderer';
import PhotoDetail from "../pages/PhotoDetail";
import "react-router-dom";

jest.mock("react-router-dom", () => {
    return {
        useLocation: () => {
            return {
                state: {
                  item: {
                    user: 'Test1',
                    largeImageURL: 'https://pixabay.com/get/g574228b852da9a2aecb82f1e915bd979194478cb4df1e4d44d6d74dbabcea4dcd7b66d2afab4d14f66452bf9f59de6d30033dc293d5b8044e515775552420e62_1280.jpg',
                    downloads: 33333,
                    views: 12312,
                    imageSize: 12343434,
                    tags: 'tags'
                  }
                }
            };
        }
    };
});

test('render PhotoDetail component', () => {  
  const component = renderer.create(    
      <PhotoDetail />    
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

