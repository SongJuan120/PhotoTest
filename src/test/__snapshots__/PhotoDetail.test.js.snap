// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PhotoDetail component 1`] = `
<div
  className="detail-main"
>
  <div
    className="user-container"
  >
    <h2>
      Name: 
      Test1
    </h2>
  </div>
  <img
    src="https://pixabay.com/get/g574228b852da9a2aecb82f1e915bd979194478cb4df1e4d44d6d74dbabcea4dcd7b66d2afab4d14f66452bf9f59de6d30033dc293d5b8044e515775552420e62_1280.jpg"
    width="80%"
  />
  <h2>
    downloads: 
    33333
  </h2>
  <h2>
    views: 
    12312
  </h2>
  <h2>
    size: 
    12343434
  </h2>
  <h2>
    Tags: 
    tags
  </h2>
</div>
`;
